x/ibc/20-transfer/module.go

```diff
@@ -238,15 +240,73 @@ func (am AppModule) OnRecvPacket(
 	if err := types.ModuleCdc.UnmarshalJSON(packet.GetData(), &data); err != nil {
 		return nil, sdkerrors.Wrapf(sdkerrors.ErrUnknownRequest, "cannot unmarshal ICS-20 transfer packet data: %s", err.Error())
 	}
-	return handlePacketDataTransfer(ctx, am.keeper, packet, data)
+	acknowledgement := FungibleTokenPacketAcknowledgement{
+		Success: true,
+		Error:   "",
+	}
+	if err := am.keeper.OnRecvPacket(ctx, packet, data); err != nil {
+		acknowledgement = FungibleTokenPacketAcknowledgement{
+			Success: false,
+			Error:   err.Error(),
+		}
+		// NOTE (cwgoes): How do we want to handle this case? Maybe we should be more lenient,
+		// it's safe to leave the channel open I think.
+
+		// TODO: handle packet receipt that due to an error (specify)
+		// the receiving chain couldn't process the transfer
+
+		// source chain sent invalid packet, shutdown our channel end
```

Comment:
We definitely don't want to do this, not sure when/where this was from, but invalid packets should not close the channel